app/src/main/java/com/sdp/movemeet/map/MainMapFragment.java

```diff
@@ -0,0 +1,208 @@
+import android.util.Log;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import androidx.annotation.RequiresApi;
+import androidx.core.app.ActivityCompat;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.sdp.movemeet.Activity.Activity;
+import com.sdp.movemeet.Activity.ActivityDescriptionActivity;
+import com.sdp.movemeet.DistanceCalculator;
+import com.sdp.movemeet.R;
+import com.sdp.movemeet.UploadActivityActivity;
+import com.sdp.movemeet.utility.ActivitiesUpdater;
+import com.sdp.movemeet.utility.LocationFetcher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.sdp.movemeet.utility.LocationFetcher.REQUEST_CODE;
+
+public class MainMapFragment extends Fragment implements GoogleMap.OnMarkerClickListener, GoogleMap.OnInfoWindowClickListener, OnMapReadyCallback, GoogleMap.OnMapClickListener {
+
+    private Location currentLocation;
+
+    private ArrayList<Activity> activities;
+
+    private FirebaseAuth fAuth = FirebaseAuth.getInstance();
+    private FirebaseUser user;
+
+    private Marker newActivityMarker;
+
+    private SupportMapFragment supportMapFragment;
+
+    private GoogleMap googleMap;
+    private FusedLocationProviderClient fusedLocationProviderClient;
+
+    private static final String TAG = "Maps TAG";
+
+    public static MainMapFragment newInstance() {
+        return new MainMapFragment();
+    }
+
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_map, container, false);
+
+        supportMapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.google_map);
+
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(supportMapFragment.getActivity());
+        LocationFetcher.fetchLastLocation(supportMapFragment, fusedLocationProviderClient, this);
+
+        ActivitiesUpdater updater = ActivitiesUpdater.getInstance();
+        activities = updater.getActivities();
+
+        user = fAuth.getCurrentUser();
+
+        return view;
+    }
+
+   /*
+    @SuppressWarnings("deprecation")
```

Comment:
I guess this was to grant the permission to fetch the user location? Why did you remove this part of code? Do we have a moment where a pop up message appears and asks the user for letting us retrieve his location?

Reply:
In the LocationFetcher,when permission is not granted, the permissions are requested. This method should just be used for additional functionalities onRequest, but I think it should be worked in the activity and not in the fragment (hence deprecated). I will take a further look on it next week.